web/Services/ApiService.cs

```diff
@@ -0,0 +1,100 @@
+﻿using BricksAndHearts.Auth;
+using BricksAndHearts.ViewModels;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+
+namespace BricksAndHearts.Services;
+
+public interface IApiService
+{
+    public Task<string> MakeApiRequestToAzureMaps(string postalCode);
+    public Task<PropertyViewModel> AutofillAddress(PropertyViewModel model);
+    public PostcodeApiResponseViewModel TurnResponseBodyToModel(string responseBody);
+}
+
+public class ApiService : IApiService
+{
+    private readonly HttpClient _client = new HttpClient();
+    private readonly ILogger<ApiService> _logger;
+    private readonly IOptions<AzureMapsOptions> _options;
+
+    public ApiService(ILogger<ApiService> logger, IOptions<AzureMapsOptions> options)
+    {
+        _logger = logger;
+        _options = options;
+    }
+
+    public async Task<string> MakeApiRequestToAzureMaps(string postalCode)
+    {
+        string uri =
+            $"https://atlas.microsoft.com/search/address/structured/json?api-version=1.0&countryCode=GB&postalCode={postalCode}&subscription-key={_options.Value.SubscriptionKey}";
+        string responseBody = String.Empty;
```

Comment:
Can you run Rider's inspector over your code?
![image](https://user-images.githubusercontent.com/35066342/181232068-5f73b4cc-4181-4654-b6ab-07b387039c4d.png)
It should come up with a bunch of small things that are usually (though not always) worth fixing, for example using var here, and giving names to the format options in the log statement below, which is important as they come through to Application Insights.

If you're not sure what any of them mean or why you might want to do them let me know.

Reply:
ok done! I've fixed up all the issues upon inspection.